test/test_torch.py

```diff
@@ -3729,6 +3729,16 @@ def test_cat(self):
 
         self.assertRaises(RuntimeError, lambda: torch.cat([]))
 
+    def test_cat_padding(self):
+        x = torch.rand(13, 1)
+        y = torch.rand(17, 2)
+        z = torch.rand(19, 3)
+
+        res1 = torch.cat((x, y, z), 0, 0)
+        self.assertEqual(res1.narrow(0, 0, 13).narrow(1, 0, 1), x, 0)
```

Comment:
do you need the 0s as the largest argument here?  Also, expected is usually the first argument given to equal, so it makes sense to reverse the other arguments.